src/lib.rs

```diff
@@ -45,38 +45,25 @@ impl World {
     fn get_state(&self, cell: &Cell) -> usize {
 
         let (x, y) = (cell.0, cell.1);
-        let top = y.checked_sub(1) != None;
-        let bot = y.checked_add(1) <= Some(self.height);
-        let right = x.checked_add(1) <= Some(self.width);
-        let left = x.checked_sub(1) != None;
+        let t = (y + self.height - 1) % self.height;
+        let b = (y + 1) % self.height;
+        let l = (x + self.width - 1) % self.width;
+        let r = (x + 1) % self.width;
```

Comment:
I don't think the smaller naming size is worth the loss of clarity here. IMO the original names are more telling of what there are for, I also tend to not like single-character variable names for things other then iterator indices. Thoughts?

Reply:
Specially because `l` and `1` look painfully alike in some fonts.

Reply:
Also, why the move away from `checked_sub()` and `checked_add()` I feel like they were more idiomatic of Rust. Thoughts?